### DISCUSSION TITLE
Modify "1-click retrieval step" page

### DISCUSSION THREAD

**Discussion by Jocky92** [pglombardo/PasswordPusher#1064] [Ideas]
For using PasswordPusher in a corporate context, it would be nice if the "1-click retrieval step" page could be customizable.

Currently the page is very simple with a text "click Here to Proceed". At first glance, this could be considered a bit odd and not reputable. It would be nice if the deposited logo could be displayed with the brand title and trand tagline. Additionally, a note about bot protection could be displayed.
A really nice solution would be to implement this with variables to be able to individually design the title, subtitle and especially the notice text.
This would make the step more comprehensible for everyone and appear a bit more confidential.

**Comment by pglombardo** [comment]
This is a great idea.  Thanks for posting this @Jocky92.

**Reply by Jocky92** [reply]
Hey @pglombardo 
i modified the _preliminary.html.erb_ by myself and added the logo and title. I am not 100% sure if everything is needed, but it looks very good. Feel free to add what ever you want^^

```
<div class="container h-100">
    <div class="d-flex flex-column min-vh-100 justify-content-center align-items-center">
        <picture>
            <% if Settings.brand.nil? || Settings.brand.light_logo.nil? %>
              <source srcset="<%= asset_path('logo-transparent-sm-bare.png') %>" media="(prefers-color-scheme: light) or (prefers-color-scheme: no-preference)">
              <source srcset="<%= asset_path('logo-transparent-sm-dark-bare.png') %>" media="(prefers-color-scheme: dark)">
              <img src="<%= asset_path('logo-transparent-sm-bare.png') %>" alt="<%= _('Password Pusher Logo') %>" style='height: 300px;' />
            <% else %>
              <source srcset="<%= Settings.brand.light_logo %>" media="(prefers-color-scheme: light) or (prefers-color-scheme: no-preference)">
              <source srcset="<%= Settings.brand.dark_logo %>" media="(prefers-color-scheme: dark)">
              <img src="<%= Settings.brand.light_logo %>" alt="<%= _('Password Pusher Logo') %>" style='height: 300px;' />
            <% end %>
        </picture>
        <span class="fs-4 display-1" style="margin-bottom: 30px;margin-top:30px;"><%= Settings.brand.title %></span>
        <p> <%=link_to _("Click Here to Proceed"), @secret_url %> </p>
    </div>
</div>
```

**Reply by pglombardo** [reply]
Thanks @Jocky92 - let me see if I can add this into the application this weekend as a new option.  Created #1116

**Comment by pglombardo** [comment]
Just an update for this request:  I've added full branding to pwpush.com.  With this you can fully customize all of the push pages.  See [here](https://docs.pwpush.com/posts/pipeline-new-features/#premium) for screenshots.

Thanks for the request @Jocky92 - hope all is well.

![premium-1-click](https://github.com/user-attachments/assets/657b5bce-6178-4ce2-9247-70a5fa079d32)
